nym-wallet/src-tauri/src/operations/mixnet/bond.rs

```diff
@@ -198,3 +210,54 @@ pub async fn get_operator_rewards(
     );
     Ok(display_coin)
 }
+
+#[tauri::command]
+pub async fn get_number_of_mixnode_delegators(
+    identity: String,
+    state: tauri::State<'_, WalletState>,
+) -> Result<usize, BackendError> {
+    let guard = state.read().await;
+    let client = guard.current_client()?;
+    let paged_delegations = client
+        .nymd
+        .get_mix_delegations_paged(identity, None, Some(20))
+        .await?;
+
+    Ok(paged_delegations.delegations.len())
+}
+
+async fn fetch_mix_node_description(
+    host: &str,
+    port: u16,
+) -> Result<NodeDescription, ReqwestError> {
+    let milli_second = Duration::from_millis(1000);
+    let client = reqwest::Client::builder().timeout(milli_second).build()?;
+    let response = client
+        .get(format!("http://{}:{}/description", host, port))
+        .send()
+        .await;
+
+    match response {
+        Ok(res) => {
+            let json = res.json::<NodeDescription>().await;
+            match json {
+                Ok(json) => Ok(json),
+                Err(e) => Err(e),
+            }
+        }
+        Err(e) => Err(e),
+    }
```

Comment:
I think you could write this match pattern as follows
```
    match response {
        Ok(res) => res.json::<NodeDescription>().await,
        Err(e) => Err(e),
    };

```

Reply:
I did try this but it was causing me an error. I'll keep as it is for now.

Reply:
hum probably because in my snippet there should be no `;` at the end of the match `block`, my bad